File: PODM/SW/pod-manager/podm-business-entities/src/main/java/com/intel/podm/business/entities/redfish/embeddables/Region.java
/*
 * Copyright (c) 2016-2018 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.intel.podm.business.entities.redfish.embeddables;

import com.intel.podm.common.types.MemoryClassification;
import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.Enumerated;

import static javax.persistence.EnumType.STRING;

@Embeddable
public class Region {
    @Column(name = "region_id")
    private String regionId;

    @Column(name = "memory_type")
    @Enumerated(STRING)
    private MemoryClassification memoryClassification;

    @Column(name = "offset_mib")
    private Integer offsetMib;

    @Column(name = "size_mib")
    private Integer sizeMib;

    public String getRegionId() {
        return regionId;
    }

    public void setRegionId(String regionId) {
        this.regionId = regionId;
    }

    public MemoryClassification getMemoryClassification() {
        return memoryClassification;
    }

    public void setMemoryClassification(MemoryClassification memoryClassification) {
        this.memoryClassification = memoryClassification;
    }

    public Integer getOffsetMib() {
        return offsetMib;
    }

    public void setOffsetMib(Integer offsetMib) {
        this.offsetMib = offsetMib;
    }

    public Integer getSizeMib() {
        return sizeMib;
    }

    public void setSizeMib(Integer sizeMib) {
        this.sizeMib = sizeMib;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }

        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        Region region = (Region) o;

        return new EqualsBuilder()
            .append(regionId, region.regionId)
            .append(memoryClassification, region.memoryClassification)
            .append(offsetMib, region.offsetMib)
            .append(sizeMib, region.sizeMib)
            .isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder()
            .append(regionId)
            .append(memoryClassification)
            .append(offsetMib)
            .append(sizeMib)
            .toHashCode();
    }
}

File: PSME/common/agent-framework/src/module/model/attributes/region.cpp
/*!
 * @section LICENSE
 *
 * @copyright
 * Copyright (c) 2015-2019 Intel Corporation
 *
 * @copyright
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * @copyright
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * @copyright
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * @section DESCRIPTION
 *
 * */

#include "agent-framework/module/model/attributes/region.hpp"
#include "agent-framework/module/constants/compute.hpp"

using namespace agent_framework::model::attribute;
using namespace agent_framework::model;

Region::Region() { }

Region::~Region() { }

json::Json Region::to_json() const {
    json::Json result = json::Json();
    result[literals::Region::REGION_ID] = get_region_id();
    result[literals::Region::MEMORY_TYPE] = get_memory_type();
    result[literals::Region::OFFSET_MIB] = get_offset_mib();
    result[literals::Region::SIZE_MIB] = get_size_mib();
    return result;
}

Region Region::from_json(const json::Json& json) {
    attribute::Region region{};
    region.set_region_id(json[literals::Region::REGION_ID]);
    region.set_memory_type(json[literals::Region::MEMORY_TYPE]);
    region.set_offset_mib(json[literals::Region::OFFSET_MIB]);
    region.set_size_mib(json[literals::Region::SIZE_MIB]);
    return region;
}
